### PR TITLE
refactor!: error handling in the vertexing algorithms

### PR DESCRIPTION
**Pull request by felix-russo**
... and lets the `Finder` consume the `Fitter` and `SeedFinder` consistently by using the `move` semantic. This was necessary in order to use `unique_ptr`s for the loggers (except for the `ImpactPointEstimator`, which is passed to both `Finder` and `Fitter`)

`BREAKING CHANGE:` When setting up the `Finder` one now needs to move `Fitter` and `SeedFinder` into it

### CODE REVIEW

File: Core/include/Acts/Vertexing/ImpactPointEstimator.hpp [hunk 1]
```diff
@@ -209,6 +213,12 @@ class ImpactPointEstimator {
   /// Configuration object
   const Config m_cfg;
 
+  /// Logging instance
+  std::shared_ptr<const Logger> m_logger;
```
Comment (felix-russo) [inline Core/include/Acts/Vertexing/ImpactPointEstimator.hpp:217]: I think this needs to be a shared pointer bc the `ImpactPointEstimator` is passed to both Finder and Fitter

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [21da8dbea382eed1733b6b0be8478e2cc5c2fe1b]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/33979344) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6485761)!

File: Core/include/Acts/Vertexing/IterativeVertexFinder.hpp [hunk 2]
```diff
@@ -75,9 +75,9 @@ class IterativeVertexFinder {
     /// @param lin Track linearizer
     /// @param sfinder The seed finder
     /// @param est ImpactPointEstimator
-    Config(const vfitter_t& fitter, Linearizer_t lin, sfinder_t sfinder,
+    Config(vfitter_t fitter, Linearizer_t lin, sfinder_t sfinder,
            const IPEstimator& est)
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/IterativeVertexFinder.hpp:79]:
```suggestion
           IPEstimator est)
```

I would make the copy explicit

File: Core/include/Acts/Vertexing/IterativeVertexFinder.hpp [hunk 3]
```diff
@@ -75,9 +75,9 @@ class IterativeVertexFinder {
...
-        : vertexFitter(fitter),
+        : vertexFitter(std::move(fitter)),
           linearizer(std::move(lin)),
           seedFinder(std::move(sfinder)),
           ipEst(est) {}
```
Comment (andiwand) [inline Core/include/Acts/Vertexing/IterativeVertexFinder.hpp:83]:
```suggestion
          ipEst(std::move(est)) {}
```

File: Core/include/Acts/Vertexing/ImpactPointEstimator.hpp (hunk 1 above)
Comment (andiwand) [reply Core/include/Acts/Vertexing/ImpactPointEstimator.hpp:217]: alternatively you could define a copy constructor which clones the logger

**Review by paulgessinger — APPROVED** [review]

**Review by andiwand — APPROVED** [review]

**Pull request merged**